core/server/server_ip.go

```diff
@@ -106,6 +145,23 @@ func runIPServer(log *zap.Logger, mtrcs *ipServerMetrics, conn *net.UDPConn, ifa
 
 		ntp.EncodePacket(&buf, &ntpresp)
 
+		var ntsresp nts.NTSPacket
+		if authenticated {
+			var cookies [][]byte
+			key := provider.Current()
+			for i := 0; i < len(ntsreq.Cookies)+len(ntsreq.CookiePlaceholders); i++ {
+				encryptedCookie, err := plaintextCookie.Encrypt(key.Value, key.Id)
```

Comment:
This looks a little surprising because none of the arguments change on each iteration. Should we call the method `EncryptWithNonce` to make what's happening here more clear?